scripts/abins/scalculatorfactory.py

```diff
@@ -16,15 +18,18 @@ class SCalculatorFactory(object):
          * SPowderSemiEmpiricalCalculator
     """
     @staticmethod
-    def init(filename=None, temperature=None, sample_form=None, abins_data=None, instrument=None,
-             quantum_order_num=None, bin_width=1.0):
+    def init(*, filename: str, temperature: float, sample_form: str,
```

Comment:
All arguments without default values following `*` become mandatory named arguments. The existing design pattern in Abins was for all arguments to be "optional", often with undefined behaviour is they are missed out. Presumably the purpose of that was to encourage clear function calls where all the arguments are named explicitly. But since Python 3.0 we no longer have to choose between those things, the `*` syntax lets us have our cake (enforce use of argument names for clarity) and eat it (make those arguments mandatory without leaving dangerous default values lying around everywhere).

https://www.python.org/dev/peps/pep-3102/

Reply:
That is interesting, thanks for clarifying